### PR TITLE
cli/command/completion: add more unit-tests

### PR DESCRIPTION
**Pull request by thaJeztah**
**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Had some of these stashed while I was working on some earlier changes; looks like I never pushed them

File: cli/command/completion/functions_test.go [hunk 1]
```diff
@@ -1,15 +1,350 @@
...
+			expDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
+		{
+			doc: "with results",
+			volumes: []*volume.Volume{
+				{Name: "volume-c"},
+				{Name: "volume-b"},
+				{Name: "volume-a"},
+			},
+			expOut:       []string{"volume-c", "volume-b", "volume-a"},
```
Comment (laurazard) [inline cli/command/completion/functions_test.go:324]: This is making me thing about whether we should order the output. I image it'd be nicer to have it alphanumerically ordered.

**Comment by laurazard** [comment]
Something else that this made me check/think about is that it'd be nice if the completion for commands that can take multiple arguments (such as `docker rm [container] [container]` didn't include arguments already added. I'll write a ticket 😅

**Review by laurazard — APPROVED** [review]
LGTM

File: cli/command/completion/functions_test.go (hunk 1 above)
Comment (thaJeztah) [reply cli/command/completion/functions_test.go:324]: Ah, yes, correct. I recalled that I (in one of my PRs) added sorting, but then noticed "it doesn't make a difference!". So somewhat purposefully made these tests show "it's not sorted, and that's ok".

We _could_ check if there's a performance difference in "pre sorting"; not sure if the shell completion code is slower at this?

**Pull request merged**